### PR TITLE
Add support for UIA accessibility headings

### PR DESCRIPTION
**Pull request by AgneLukoseviciute**
Related to: https://github.com/microsoft/react-native-windows/issues/8452
When accessibilityRole is set to "header" in RN-core, the native values that are updated (Heading for Android and accessibilityTrait for iOS) are the same values that used by screen readers to determine if some text is a heading. On Windows, screen readers check the UIA Heading Level to determine if something is a heading -- this PR sets the UIA Heading Level to a default level 2 whenever accessibilityRole="header" in Text. This will match the behavior in RN-core, although we might want to create a windows-specific prop down the road to allow for control over the exact heading level passed to UIA. 

Also included is an update to automationProperties.localizedControlTypeProperty when setting accessibilityRole in Text as a slight improvement for https://github.com/microsoft/react-native-windows/issues/8191 until we decide on a workaround for custom automation peer functionality for Text. 

**Testing:** Tested using Narrator and Accessibility Insights For Windows. When accessibilityRole="header" is set on a Text component Narrator Scan mode announces the heading as expected and Accessibility Insights shows the localizedControlTypeProperty as "header". 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8516)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Views/TextViewManager.cpp [hunk 1]
```diff
@@ -288,6 +288,22 @@ bool TextViewManager::UpdateProperty(
...
+  } else if (propertyName == "accessibilityRole") {
+    if (propertyValue.Type() == winrt::Microsoft::ReactNative::JSValueType::String) {
+      const std::string &role = propertyValue.AsString();
+      auto value = asHstring(propertyValue);
+      auto boxedValue = winrt::Windows::Foundation::PropertyValue::CreateString(value);
+
+      textBlock.SetValue(winrt::AutomationProperties::LocalizedControlTypeProperty(), boxedValue);
+      if (role == "header") {
+        xaml::Automation::AutomationProperties::SetHeadingLevel(
+            textBlock, winrt::Peers::AutomationHeadingLevel::Level2);
```
Comment (ksiler) [inline vnext/Microsoft.ReactNative/Views/TextViewManager.cpp:300]: Is there a reason why you chose heading level 2 as the default as opposed to level 1?

Comment (AgneLukoseviciute) [reply vnext/Microsoft.ReactNative/Views/TextViewManager.cpp:300]: @ksiler The team currently blocked by this issue was needing Level2 headings specifically, that's really the main reason why. Also, [looks like Adaptive Cards has chosen level 2 as the default](https://github.com/microsoft/AdaptiveCards/blob/main/specs/DesignDiscussions/TextHeadings.md) -- I'm checking in with them about that.

File: vnext/Microsoft.ReactNative/Views/TextViewManager.cpp [hunk 2]
```diff
@@ -288,6 +288,22 @@ bool TextViewManager::UpdateProperty(
...
+      } else {
+        textBlock.ClearValue(winrt::AutomationProperties::HeadingLevelProperty());
+      }
+    } else if (propertyValue.IsNull()) {
+      textBlock.ClearValue(winrt::AutomationProperties::LocalizedControlTypeProperty());
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Views/TextViewManager.cpp:305]: You probably also want to clear the heading level property if the `accessibilityRole` changes from `header` to `null`.
```suggestion
      textBlock.ClearValue(winrt::AutomationProperties::LocalizedControlTypeProperty());
      textBlock.ClearValue(winrt::AutomationProperties::HeadingLevelProperty());
```

**Review by jonthysell — APPROVED** [review]

File: vnext/Microsoft.ReactNative/Views/TextViewManager.cpp (hunk 2 above)
Comment (AgneLukoseviciute) [reply vnext/Microsoft.ReactNative/Views/TextViewManager.cpp:305]: good catch, thank you!

**Pull request merged**